enhancements/installer/ibm-cloud-for-power-vs-ipi.md

```diff
@@ -0,0 +1,278 @@
+10. The cluster teardown can make use of the existing [go client for Power][power-go-client] via
+`openshift-install destroy cluster`.
+
+### User Stories
+
+- A user of this installer would like to see a cluster running within an hour of kicking off the installer.
+- The expectation is that this is fully automated.
+
+### Implementation Details/Notes/Constraints
+
+#### CoreOS OVA Considerations
+There were two main implementation ideas for making the CoreOS OVA boot image available to the Virtual Server instance.
+
+1. Import the Image from a Cloud Store
+This is the implementation we believe is best, since it works most similarly to other cloud environments. In this setup,
+the CoreOS build pipeline would publish an OVA image to an IBM Cloud Object Storage (COS) bucket. Then, we will use the
+regular IBM Cloud APIs to import the image to the instance. The drawback to this approach is the the current
+implementation is very slow. In its current state, it takes about 30 minutes to import an 120 Gb boot image. This is too
+long to meet our target of being able to install a whole cloud within the hour. One of the reasons this is slow is
+because we cannot do the image resizing after the copy, since it's writing directly to a sized disk. Once imported,
+creating the nodes is fast, since the other disks are flashed from the snapshot. For now, we asked the team to look into
+meeting the requirement that importing the image can take 5 minutes or less. Another bit we're working on is defining
+the implementation details for how the image will be imported from the COS bucket to the PowerVS service.
+Currently, the terraform API doesn't support importing an image from bucket without key authentication. We are looking
+into if it is possible to update the terraform provider to do this, since IBM Cloud Object Storage does support public
+buckets that shouldn't need key authentication.
+
+2. Published CoreOS Images to the IBM Cloud Pre-Authorized Default Images
+This approach would involve outputting our CoreOS OVA images in a queue where they can be pre-loaded into the IBM Cloud
+so that they are available as soon as the Virtual Server instance is available. This is significantly quicker than the
+method above but has several drawbacks. For one, these default images need to be pre-authorized in order for them to be
+added to the IBM Cloud. This is currently a manual process owned by the Power VS team, so it cannot be completed by a
+user. Additionally, while we could build automation to solve this, it's unclear how would allow the user to point to
+the correct CoreOS image in the default list, since they would have to know the image name and details before their
+instance is created. Finally, we don't want to muddy up the default image list with an ever growing list of CoreOS
+options. This pruning would need to be automated some other mechanism would need to be in place to protect the user
+experience.
```

Comment:
I think this is similar to how it is done on GCP. We have images in buckets, and then the installer knows a specific string to use for a 4.6/4.7/4.8 cluster respectively. I think this would likely be a preferred approach.

Does adding them to this registry mean they are only available to PowerVS, we were having similar conversations with the IBM VPC team, just wondering if there's a common solution that could be achieved here

Reply:
The idea here is that as part of the RHCOS pipeline, the ova images will be created and pushed to a specific COS bucket in IBMCloud. the replication to multiple regions has to be done manually as there is not automated API to do that. with this PR in the coreos-assembler: https://github.com/coreos/coreos-assembler/pull/2200, the RHCOS pipeline will have the ability to build and push both PowerVS ovas and IBMCloud qcows to COS buckets in IBMCloud. so in that sense RHCOS will support building and pushing images for both IBMCloud and PowerVS. The only thing that needs to be changed is pushing to separate COS buckets for both and not use the same one.